gtsam/hybrid/HybridGaussianConditional.cpp

```diff
@@ -28,56 +28,47 @@
 #include <gtsam/linear/GaussianFactorGraph.h>
 
 namespace gtsam {
-HybridGaussianFactor::FactorValuePairs GetFactorValuePairs(
+/* *******************************************************************************/
+HybridGaussianConditional::ConstructorHelper::ConstructorHelper(
     const HybridGaussianConditional::Conditionals &conditionals) {
-  auto func = [](const GaussianConditional::shared_ptr &conditional)
-      -> GaussianFactorValuePair {
+  negLogConstant = std::numeric_limits<double>::infinity();
+
+  auto func =
+      [&](const GaussianConditional::shared_ptr &c) -> GaussianFactorValuePair {
     double value = 0.0;
-    // Check if conditional is pruned
-    if (conditional) {
-      // Assign log(\sqrt(|2πΣ|)) = -log(1 / sqrt(|2πΣ|))
-      value = conditional->negLogConstant();
+    if (c) {
+      if (frontals.empty()) {
```

Comment:
Is this needed? Won't a conditional always have a frontal variable?

Reply:
No use copying frontals if they've already been copied.